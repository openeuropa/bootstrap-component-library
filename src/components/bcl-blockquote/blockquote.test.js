import { renderTwigFileAsNode } from "@openeuropa/bcl-test-utils";
import demoData from "@openeuropa/bcl-data-blockquote/data";

const template = "@oe-bcl/bcl-blockquote/blockquote.html.twig";
const render = (params) => renderTwigFileAsNode(template, params, true);

describe("OE - Blockquote", () => {
  test(`renders correctly`, () => {
    expect.assertions(1);

    return expect(render({ ...demoData })).resolves.toMatchSnapshot();
  });

  test(`renders correctly with no cite`, () => {
    expect.assertions(1);

    return expect(render({ ...demoData, cite: "" })).resolves.toMatchSnapshot();
  });

  test(`renders correctly with no attribution`, () => {
    expect.assertions(1);

    return expect(
      render({ ...demoData, attribution: "" })
    ).resolves.toMatchSnapshot();
  });

  test(`renders correctly with text aligned center`, () => {
    expect.assertions(1);

    return expect(
      render({ ...demoData, alignment: "center" })
    ).resolves.toMatchSnapshot();
  });

  test(`renders correctly with text aligned right`, () => {
    expect.assertions(1);

    return expect(
      render({ ...demoData, alignment: "right" })
    ).resolves.toMatchSnapshot();
  });
});
