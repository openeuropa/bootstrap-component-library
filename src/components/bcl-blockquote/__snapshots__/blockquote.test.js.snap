// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OE - Blockquote renders correctly 1`] = `
<jest>
  <figure
    class="text-left bg-lighter p-2 px-2-5 border-4 border-secondary border-start"
  >
    <blockquote
      class="blockquote mb-2"
    >
      <p>
        A well-known quote, contained in a blockquote element.
      </p>
    </blockquote>
    <figcaption
      class="blockquote-footer m-0"
    >
      Someone famous in    
      <cite
        title="Source title"
      >
        Source title
      </cite>
    </figcaption>
  </figure>
</jest>
`;

exports[`OE - Blockquote renders correctly with no attribution 1`] = `
<jest>
  <figure
    class="text-left bg-lighter p-2 px-2-5 border-4 border-secondary border-start"
  >
    <blockquote
      class="blockquote mb-2"
    >
      <p>
        A well-known quote, contained in a blockquote element.
      </p>
    </blockquote>
  </figure>
</jest>
`;

exports[`OE - Blockquote renders correctly with no cite 1`] = `
<jest>
  <figure
    class="text-left bg-lighter p-2 px-2-5 border-4 border-secondary border-start"
  >
    <blockquote
      class="blockquote mb-2"
    >
      <p>
        A well-known quote, contained in a blockquote element.
      </p>
    </blockquote>
    <figcaption
      class="blockquote-footer m-0"
    >
      Someone famous in  
    </figcaption>
  </figure>
</jest>
`;

exports[`OE - Blockquote renders correctly with text aligned center 1`] = `
<jest>
  <figure
    class="text-center bg-lighter p-2 px-2-5 border-4 border-secondary border-top"
  >
    <blockquote
      class="blockquote mb-2"
    >
      <p>
        A well-known quote, contained in a blockquote element.
      </p>
    </blockquote>
    <figcaption
      class="blockquote-footer m-0"
    >
      Someone famous in    
      <cite
        title="Source title"
      >
        Source title
      </cite>
    </figcaption>
  </figure>
</jest>
`;
